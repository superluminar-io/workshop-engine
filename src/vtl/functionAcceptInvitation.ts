export const request = `
  {
    "version" : "2017-02-28",
    "operation" : "UpdateItem",
    "key" : {
      "id": $util.dynamodb.toDynamoDBJson($ctx.arguments.inviteId)
    },
    "update": {
      "expression" : "SET attendees = list_append(attendees, :attendees)",
      "expressionValues": {
        ":attendees": $util.dynamodb.toDynamoDBJson([$ctx.arguments.emailAddress])
      }
    }
  }
`;

export const response = `
  $util.toJson($ctx.result)
`;
