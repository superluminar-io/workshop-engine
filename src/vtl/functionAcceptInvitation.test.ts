import Parser from 'appsync-template-tester';
import { request, response } from './functionAcceptInvitation';

describe('request mapping', () => {
  it('should match snapshot', () => {
    const parser = new Parser(request);

    const context = {
      arguments: {
        inviteId: 'c87c4461-ad5a-45ce-bc42-87e9b10e1bab',
        emailAddress: 'me@xample.com',
      },
    };

    expect(parser.resolve(context)).toMatchSnapshot();
  });
});

describe('response mapping', () => {
  it('should match snapshot', () => {
    const parser = new Parser(response);

    const context = {
      result: {
        any: 'data',
      },
    };

    expect(parser.resolve(context)).toMatchSnapshot();
  });
});
