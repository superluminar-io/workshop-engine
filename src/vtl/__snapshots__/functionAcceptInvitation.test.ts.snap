// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request mapping should match snapshot 1`] = `
Object {
  "key": Object {
    "id": Object {
      "S": "c87c4461-ad5a-45ce-bc42-87e9b10e1bab",
    },
  },
  "operation": "UpdateItem",
  "update": Object {
    "expression": "SET attendees = list_append(attendees, :attendees)",
    "expressionValues": Object {
      ":attendees": Object {
        "L": Array [
          Object {
            "S": "me@xample.com",
          },
        ],
      },
    },
  },
  "version": "2017-02-28",
}
`;

exports[`response mapping should match snapshot 1`] = `
Object {
  "any": "data",
}
`;
